playground/backend/internal/code_processing/code_processing.go

```
@@ -235,6 +239,52 @@ func cancelCheck(ctx context.Context, pipelineId uuid.UUID, cancelChannel chan b
 	}
 }
 
+// readLogFile reads logs from the log file and keeps it to the cache.
+func readLogFile(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycle, pipelineId uuid.UUID, stopReadLogsChannel, finishReadLogChannel chan bool) {
+	ticker := time.NewTicker(500 * time.Millisecond)
+	for {
+		select {
+		// in case of timeout or cancel
+		case <-ctx.Done():
+			finishReadLogFile(ctx, ticker, cacheService, lc, pipelineId)
+			return
+		// in case of pipeline finish successfully or has error on the run step
+		case <-stopReadLogsChannel:
+			finishReadLogFile(ctx, ticker, cacheService, lc, pipelineId)
+			finishReadLogChannel <- true
+			return
+		case <-ticker.C:
+			if err := writeLogsToCache(ctx, cacheService, lc, pipelineId); err != nil {
+				return
+			}
+		}
+	}
+}
+
+// finishReadLogFile is used to read logs file for the last time
+func finishReadLogFile(ctx context.Context, ticker *time.Ticker, cacheService cache.Cache, lc *fs_tool.LifeCycle, pipelineId uuid.UUID) {
+	ticker.Stop()
+	_ = writeLogsToCache(ctx, cacheService, lc, pipelineId)
+}
+
+// writeLogsToCache write all logs from the log file to the cache.
+// If log file doesn't exist, return nil.
+// If log file exists, read all from the log file and keep it to the cache using.
+// If some error occurs, log the error and return err.
+func writeLogsToCache(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycle, pipelineId uuid.UUID) error {
+	if _, err := os.Stat(lc.GetAbsoluteLogFilePath()); os.IsNotExist(err) {
+		return nil
+	}
+	logs, err := os.ReadFile(lc.GetAbsoluteLogFilePath())
+	if err != nil {
+		logger.Errorf("%s: writeLogsToCache(): error during read from logs file: %s", pipelineId, err.Error())
+		return err
+	}
+	_ = cacheService.SetValue(ctx, pipelineId, cache.Logs, string(logs))
```

why do you ignore errors here?

---

Changed.